src/esp/physics/bullet/BulletArticulatedObject.cpp

```diff
@@ -132,6 +139,47 @@ bool BulletArticulatedObject::initializeFromURDF(
     bWorld_->addMultiBody(btMultiBody_.get());
     btMultiBody_->setCanSleep(true);
 
+    bFixedObjectShape_ = std::make_unique<btCompoundShape>();
+
+    // By convention, fixed links in the URDF are assigned Noncollidable, and we
+    // then insert corresponding fixed rigid bodies with group Static.
+    // Collisions with a fixed rigid body are cheaper than collisions with a
+    // fixed link, due to problems with multibody sleeping behavior.
+    {
+      auto* col = mb->getBaseCollider();
+      if (col->getBroadphaseHandle()->m_collisionFilterGroup ==
+          int(CollisionGroup::Noncollidable)) {
+        bFixedObjectShape_.get()->addChildShape(col->getWorldTransform(),
+                                                col->getCollisionShape());
+      }
+
+      for (int m = 0; m < mb->getNumLinks(); m++) {
+        btMultiBodyLinkCollider* col = mb->getLink(m).m_collider;
+        if (col) {
+          if (col->getBroadphaseHandle()->m_collisionFilterGroup ==
+              int(CollisionGroup::Noncollidable)) {
+            bFixedObjectShape_.get()->addChildShape(col->getWorldTransform(),
+                                                    col->getCollisionShape());
+          }
+        }
+      }
+
+      if (bFixedObjectShape_.get()->getNumChildShapes()) {
+        btRigidBody::btRigidBodyConstructionInfo info =
+            btRigidBody::btRigidBodyConstructionInfo(0.f, nullptr,
+                                                     bFixedObjectShape_.get());
+        bFixedObjectRigidBody_ = std::make_unique<btRigidBody>(info);
+        BulletDebugManager::get().mapCollisionObjectTo(
+            bFixedObjectRigidBody_.get(),
+            "URDF, " + u2b.getModel()->m_name + ", fixed body");
+        bWorld_->addRigidBody(
```

Comment:
We'll also want to add this new fixed base object to `collisionObjToObjIds_` map for collision queries. It should then be removed in the destructor.